src/views/Home.jsx

```diff
@@ -1,11 +1,34 @@
 import './Home.css';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { createNewList } from '../api/firebase';
+
+/** Home component that redirects a user to the List view if there is already a list created.
+ * If the user doesn't already have a list, a user can create one to be saved to Firestore and be redirected to the List view. */
+
+export function Home({ listToken, setListToken }) {
+	const navigate = useNavigate();
+
+	useEffect(() => {
+		if (listToken) {
+			navigate('/list');
+		}
+	}, []);
+
+	const createNewToken = async () => {
+		const newToken = generateToken();
+		setListToken(newToken);
```

Comment:
one thing to consider in the future would be handling a case where the `createNewList` call fails. I haven't tested it locally (so it might totally not work), but one version of this function might look like this:
```javascript
const createNewToken = async () => {
  const newToken = generateToken();
  createNewList(newToken)
    .then(() => {
      setListToken(newToken);
      navigate('/list');
    })
    .catch((error) => {
      // should we tell the user something went wrong?
     });
}
```

All that said, I believe @daredtech has some feedback about how this interaction between `listToken` and `navigate` is working, so I'd encourage you to take a look at that.